### ISSUE TITLE
Installing datalad subdatasets on a mounted cifs filesystem does not work

### ISSUE THREAD

**Issue by JohannesWiesner** [datalad/datalad#7086]
### What is the problem?

I am having trouble with installing datalad subdatasets on a mounted cifs filesystem. This issue is probably related to #6917. I tried to make sure that I am following the suggestions from this issue by configuring

`git config annex.pidlock true`  and  `git config annex.dbdir $HOME/cache/git-annex`

### What steps will reproduce the problem?

Following the steps [from this datalad tutorial](https://handbook.datalad.org/en/latest/beyond_basics/101-149-copyfile.html#the-basics-of-copy-file) results in:

```bash
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing$ datalad clone https://github.com/datalad-datasets/human-connectome-project-openaccess.git hcp
install(ok): /zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp (dataset)                                                                            
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing$ cd hcp
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp$ git config annex.pidlock true
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp$ git config annex.dbdir $HOME/cache/git-annex
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp$ datalad get -n -r HCP1200/130*
[INFO   ] git-annex: Not initialized.                                           
[INFO   ] init: 1 failed 
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp$ h exitcode 1 under /zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp/HCP1200/130013
```

### DataLad information

datalad version: `0.17.6`
git-annex version: `10.20220822-g84f1875`
OS: Linux Mint 19.3 "Tricia"

**Comment by bpoldrack** [comment]
@JohannesWiesner: Thanks for reporting the issue!

While I have no immediate idea really, at the very least it looks like git-annex' error reporting is a bit strange here:

```
[INFO   ] git-annex: Not initialized.                                           
[INFO   ] init: 1 failed
```

We need to double-check we are not swallowing part of it.

**Comment by JohannesWiesner** [comment]
Yes, the reporting is quite sparse. My personal hypothesis is, that the issue from #6917 is just happening in all of the subdatasets, since they all have their own annex directory? Of course, for this case it would not be possible to specify

```
it config annex.pidlock true
it config annex.dbdir $HOME/cache/git-annex
```

for each of those manually. I guess you would have to find a way to automatize this and to propagate these settings to all following commands after you install the main dataset.

**Comment by yarikoptic** [comment]
FWIW: @JohannesWiesner I am curious -- your CIFS, does it support symlinks? ([quick google pointed me to the fact that it might be possible](https://www.jeffgeerling.com/blog/2021/making-sure-symlinks-work-on-cifssmb-mounted-shares))?  my thinking is that if it does, then if there is a way to "freeze/thaw" (make read-only/read-write) as well -- possibly new functionality of git-annex to provide custom freeze/thawing could be used.  An example of such is briefly outlined in step 2 of https://dbic-handbook.readthedocs.io/en/latest/mri/dataaccess.html#discovery-filesystem .

**Comment by yarikoptic** [comment]
or am I completely off in that you are not "fighting" adjusted branches here at all yet?

**Comment by JohannesWiesner** [comment]
> FWIW: @JohannesWiesner I am curious -- your CIFS, does it support symlinks? ([quick google pointed me to the fact that it might be possible](https://www.jeffgeerling.com/blog/2021/making-sure-symlinks-work-on-cifssmb-mounted-shares))? my thinking is that if it does, then if there is a way to "freeze/thaw" (make read-only/read-write) as well -- possibly new functionality of git-annex to provide custom freeze/thawing could be used. An example of such is briefly outlined in step 2 of https://dbic-handbook.readthedocs.io/en/latest/mri/dataaccess.html#discovery-filesystem .

Yes it does. Of course, these links only work when you access the file system from a Linux based OS. For example:

```bash
(base) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks$ mkdir src
(base) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks$ mkdir tgt
(base) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks$ cd src
(base) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks/src$ cat > src.txt
Hello World!
(base) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks/src$ ln -s /zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks/src/src.txt /zi-flstorage/group_csp/analyses/johannes.wiesner/testing/test_symlinks/tgt/tgt.txt
```

works fine. When you try to open the symlink from Windows this will fail.

**Comment by JohannesWiesner** [comment]
> or am I completely off in that you are not "fighting" adjusted branches here at all yet?

Sorry, I am not sure exactly what you mean with that? 

>An example of such is briefly outlined in step 2 of https://dbic-handbook.readthedocs.io/en/latest/mri/dataaccess.html#discovery-filesystem  

I am a little bit confused how I would have to replace those lines of code for my example?

```
it config --global annex.thawcontent-command '/dartfs/rc/lab/D/DBIC/DBIC/archive/bin-annex/thaw-content %path'
git config --global annex.freezecontent-command = /dartfs/rc/lab/D/DBIC/DBIC/archive/bin-annex/freeze-content %path
```

**Comment by bpoldrack** [comment]
> Yes, the reporting is quite sparse. My personal hypothesis is, that the issue from #6917 is just happening in all of the subdatasets, since they all have their own annex directory? Of course, for this case it would not be possible to specify
>
>```
>git config annex.pidlock true
>git config annex.dbdir $HOME/cache/git-annex
>```

You may be right about this, @JohannesWiesner. Did you also try to set those configs with `--global`? I'd be surprised if annex wouldn't respect that.

**Comment by yarikoptic** [comment]
> > or am I completely off in that you are not "fighting" adjusted branches here at all yet?
> 
> Sorry, I am not sure exactly what you mean with that?

https://git-annex.branchable.com/design/adjusted_branches/ which are used whenever working on "crippled filesystem". Just type `git branch` and see if you are in `master` or `adjusted/master`


> I am a little bit confused how I would have to replace those lines of code for my example?
> 
> ```
> git config --global annex.thawcontent-command '/dartfs/rc/lab/D/DBIC/DBIC/archive/bin-annex/thaw-content %path'
> git config --global annex.freezecontent-command = /dartfs/rc/lab/D/DBIC/DBIC/archive/bin-annex/freeze-content %path
> ```

that would depend on how we could ensure content to be "forced" to become "Read-only", and answer to

> works fine. When you try to open the symlink from Windows this will fail.

will you want to access them from Windows? if yes -- then adjusted branches I guess is the only way but there is lots of nuances and issues (especially when dealing with subdatasets) about that approach.

**Comment by JohannesWiesner** [comment]
> > Yes, the reporting is quite sparse. My personal hypothesis is, that the issue from #6917 is just happening in all of the subdatasets, since they all have their own annex directory? Of course, for this case it would not be possible to specify
> > ```
> > git config annex.pidlock true
> > git config annex.dbdir $HOME/cache/git-annex
> > ```
> 
> You may be right about this, @JohannesWiesner. Did you also try to set those configs with `--global`? I'd be surprised if annex wouldn't respect that.

Good idea! I will let you know next week :)

**Comment by yarikoptic** [comment]
please do

```
git -C /zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp -c diff.ignoreSubmodules=none annex init -c annex.dotfiles=true
```
and share the output

**Comment by JohannesWiesner** [comment]
This gives me:

```bash
(dl) johannes.wiesner@zilxhp01:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp$ git -C /zi-flstorage/group_csp/analyses/johannes.wiesner/testing/hcp -c diff.ignoreSubmodules=none annex init -c annex.dotfiles=true
init  
  Initialization prevented by .noannex file (remove the file to override)

git-annex: Not initialized.
failed
init: 1 failed
```

**Comment by bpoldrack** [comment]
This is interesting, since we failed to notice so far. So, the superdataset "works" not b/c of correct settings, but because it simply is no annex repository (it probably only references subdatasets and has no annexed content itself).

However, this is not true for the subdataset `HCP1200/130013`. So, this doesn't give us much of an insight into what's going on.

**Comment by bpoldrack** [comment]
@JohannesWiesner:

You could try to directly clone this subdataset:

```
> git clone http://store.datalad.org/78a/4404e-2aa7-11ea-8c3f-002590496000 testds
> cd testds
> git annex init
```

**Comment by JohannesWiesner** [comment]
This gives me:

```bash
(csp_wiesner_johannes) johannes.wiesner@zilxap02:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing$ git config --global annex.pidlock true
(csp_wiesner_johannes) johannes.wiesner@zilxap02:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing$ git config --global annex.dbdir $HOME/cache/git-annex
(csp_wiesner_johannes) johannes.wiesner@zilxap02:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing$ git clone http://store.datalad.org/78a/4404e-2aa7-11ea-8c3f-002590496000 testds
Cloning into 'testds'...
Fetching objects: 1566, done.
(csp_wiesner_johannes) johannes.wiesner@zilxap02:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing$ cd testds
(csp_wiesner_johannes) johannes.wiesner@zilxap02:/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/testds$ git annex init
init  
  Detected a filesystem where Sqlite does not work.

  (/zi-flstorage/group_csp/analyses/johannes.wiesner/testing/testds/.git/annex: openTempFile template locktmp: does not exist (No such file or directory))

  To work around this problem, you can set annex.dbdir to a directory on another filesystem.

  For example: git config annex.dbdir $HOME/cache/git-annex

git-annex: Not initialized.
failed
```
so we basically ending up with #6917, strange...

**Comment by bpoldrack** [comment]
@JohannesWiesner 

Ok, so to sum up: This is not about subdatasets (the seemingly working top-level is a plain git repository). It's annex not being able to initialize on this file system. I have no idea of cifs and what's the problem for annex here, but I don't think there's anything we can do at the moment on datalad's end. Maybe you want to consider filing this with [git-annex](https://git-annex.branchable.com/) instead?

Or maybe @joeyh can comment on this right away?

**Comment by bpoldrack** [comment]
As far as datalad is concerned, it's just another instance, where we should be more verbose in relaying `git-annex` output, so getting to this point is less of a hassle.
